src/pdl/pdl_ast.py

```diff
@@ -462,6 +462,9 @@ class CodeBlock(LeafBlock):
     ]
     """Programming language of the code.
     """
+    requirements: Optional[str] = None
```

Comment:
This should be only enabled for `lang: python`. Should we check for it?

Reply:
this issue seems to indicate that we could do it, but only with a custom model validator https://github.com/pydantic/pydantic/issues/10252

Reply:
ok i added a model validator. thanks for the good catch!

Reply:
I was going to suggest Pydantic validator as well.  Also wondering if `requirements` should actually be `Optional[str | List[str]`?

Reply:
very good! done